frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -275,6 +276,14 @@ export const getCommonInitialValues = (
   return commonInitialValues;
 };
 
+export const getIconInitialValues = (editAppResource: K8sResourceKind) => {
+  const runtimeLabel = editAppResource?.metadata?.labels?.['app.openshift.io/runtime'];
+  const runtimeIcon = runtimeLabel && hasIcon(runtimeLabel) ? runtimeLabel : null;
```

Comment:
Now that there is a default (see my other comment)... we probably should have it here as well. Maybe as a shared constant in like the `imagestream-utils` file or something. 

My only concern is I don't want to default to the `openshift` icon if it has an icon and for some reason we don't support it. Maybe the logic is:

- if `runtimeLabel` is not found => `openshift` icon
- if `runtimeLabel` is found BUT `hasIcon` is false => `null`

Reply:
I changed the implementation to this and keep the default on null. So if the user do not select another icon we do not need to save the label. If the user selects one we save the additional label. So we can change the default behavior at any time. OK?

```jsx
    const icon = getIcon(value || 'openshift');
    return icon ? (
      <Icon label={icon.label} url={icon.url} />
    ) : (
      <span className="btn-dropdown__item--placeholder">{placeholder}</span>
    );
```

Reply:
We could fake the default... I think it's helpful for the user to see the icon. If it's null we could set the view to render the OpenShift icon instead of the placeholder. But I think we lose the fidelity of informing when there just isn't an icon... maybe check with UX. Maybe this is a moot point?